.github/workflows/test.yml

```diff
@@ -63,10 +69,37 @@ jobs:
         uses: Swatinem/rust-cache@v2
       - name: Setup coverage env
         uses: taiki-e/install-action@cargo-llvm-cov
+      # Python setup
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Load cached Poetry installation
+        id: cached-poetry
+        uses: actions/cache@v3
+        with:
+          path: ~/.local
+          key: poetry-${{ runner.os }}
+      - name: Install Poetry
+        if: steps.cached-poetry.outputs.cache-hit != 'true'
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+      - name: pull kakarot submodule and setup
+        run:
+          make setup
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
```

Comment:
the venv is loaded after `make setup`?

Reply:
what env variable is make setup using?

Reply:
i do not think i am doing this right, i was emulating the kakarot ci and i think there's output in the runner that it isn't setup correctly. will link to it here when i find it again